virttest/test_setup/__init__.py

```diff
@@ -834,6 +838,14 @@ def __init__(self, params=None):
                 self.force_create = True
         self.bridge_manager = utils_net.Bridge()
 
+    def _check_firewall_tool(self):
+        cmd = "find /usr/lib/modules/ -name iptable_filter.ko.xz"
+        result = process.system_output(cmd)
+        if result:
+            return "iptables"
+        else:
+            return "nft"
+
```

Comment:
Even if iptable_filter can be found, that doesn't mean Firewalld uses it.

To check the Firewalld's backend, is to check its config file.
```
$ grep "^FirewallBackend" /etc/firewalld/firewalld.conf
FirewallBackend=nftables
```

I would suggest using firewall-cmd to add/del rules, firewall-cmd is the front, with this we don't need to care about the backend, and it can handle common use cases.

Reply:
The firewalld utility can only handle simple firewall use cases and cannot set up complex firewalls. sometimes It cannot provide equivalent iptables/nft commands. Also, I'm not sure if all systems have firewalld packages available. using firewalld instead may affect the stability of the framework.